src/BlockTypes/ProductQuery.php

```diff
@@ -120,9 +120,11 @@ public function update_query( $pre_render, $parsed_block ) {
 	 * @param WP_REST_Request $request Request.
 	 */
 	public function update_rest_query( $args, $request ) {
-		$on_sale_query = $request->get_param( '__woocommerceOnSale' ) !== 'true' ? array() : $this->get_on_sale_products_query();
-		$orderby_query = $this->get_custom_orderby_query( $request->get_param( 'orderby' ) );
-		$tax_query     = $this->get_product_attributes_query( $request->get_param( '__woocommerceAttributes' ) );
+		$orderby        = $request->get_param( 'orderby' );
+		$woo_attributes = $request->get_param( '__woocommerceAttributes' );
+		$on_sale_query  = $request->get_param( '__woocommerceOnSale' ) === 'true' ? $this->get_on_sale_products_query() : array();
+		$orderby_query  = isset( $orderby ) ? $this->get_custom_orderby_query( $orderby ) : array();
+		$tax_query      = isset( $woo_attributes ) ? $this->get_product_attributes_query( $woo_attributes ) : array();
```

Comment:
As `get_product_attributes_query()` accepts array, should we change this to ensure we won't get any errors, as the request param may be changed by other actors in the merchants' sites? It's not likely to be changed but we can never be sure about that.

```suggestion
		$tax_query      = is_array( $woo_attributes ) ? $this->get_product_attributes_query( $woo_attributes ) : array();
```